src/ILLink.Shared/TrimAnalysis/HandleCallAction.cs

```diff
@@ -666,8 +666,16 @@ GenericParameterValue genericParam
 				break;
 
 			case IntrinsicId.None:
-				methodReturnValue = MultiValueLattice.Top;
-				return false;
+				// Verify the argument values match the annotations on the parameter definition
+				if (requiresDataFlowAnalysis) {
+					if (!calledMethod.IsStatic ()) {
+						_requireDynamicallyAccessedMembersAction.Invoke (instanceValue, GetMethodThisParameterValue (calledMethod));
+					}
+					for (int argumentIndex = 0; argumentIndex < argumentValues.Count; argumentIndex++) {
+						_requireDynamicallyAccessedMembersAction.Invoke (argumentValues[argumentIndex], GetMethodParameterValue (calledMethod, argumentIndex));
+					}
+				}
+				break;
```

Comment:
Now that we always return `true`, we can stock handling the `IntrinsicId.None` case when this is called from the analyzer in `TrimAnalysisVisitor`.

Reply:
I would also prefer if we could use the new shared logic for `IntrinsicId.None` on the linker side. The linker has some extra handling in the `default` case that we would need to separate out, but if we are sharing this code we should remove the corresponding code from linker/analyzer if possible.

Reply:
BTW: The method still returns `false` in some cases, like for example for `Array.Empty<T>` since we don't have a good way (yet) to handle it in shared code (we don't have a way to handle generic instantiations to be more precise).